nearcache/nearcache_it_test.go

```diff
@@ -0,0 +1,327 @@
+	"github.com/hazelcast/hazelcast-go-client/internal/it"
+	"github.com/hazelcast/hazelcast-go-client/nearcache"
+)
+
+const (
+	nearCacheDefaultRecordCount = 1000
+)
+
+// TestWhenGetIsUsedThenNearCacheShouldBePopulated checks that the Near Cache is populated when Get is used.
+// And also the NearCacheStats are calculated correctly.
+func TestWhenGetIsUsedThenNearCacheShouldBePopulated(t *testing.T) {
+	// ported from: com.hazelcast.internal.nearcache.impl.AbstractNearCacheBasicTest#whenGetIsUsed_thenNearCacheShouldBePopulated()
+	clientCacheNearCacheBasicSlowRunner(t, func(tcx *it.NearCacheTestContext, size int64, valueFmt string) {
+		tcx.PopulateNearCacheWithGet(size, valueFmt)
+	})
+}
+
+func TestNearCacheGet(t *testing.T) {
+	// port of: com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest#testGetAsync
+	tcx := newNearCacheMapTestContext(t, nearcache.InMemoryFormatObject, false)
+	tcx.Tester(func(tcx it.MapTestContext) {
+		m := tcx.M
+		t := tcx.T
+		const size = int64(1009)
+		ctx := context.Background()
+		populateMap(tcx, size)
+		populateNearCache(tcx, size)
+		// generate near cache hits
+		for i := int64(0); i < size; i++ {
+			v := it.MustValue(m.Get(ctx, i))
+			require.Equal(t, i, v)
+		}
+		stats := m.LocalMapStats().NearCacheStats
+		require.Equal(t, size, stats.Hits)
+		require.Equal(t, size, stats.OwnedEntryCount)
+	})
+}
+
+type mapTestCase struct {
+	name string
+	f    func(context.Context, it.MapTestContext, int64)
+}
+
+func TestAfterRemoveNearCacheIsInvalidated(t *testing.T) {
+	// port of: com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest#testAfterRemoveNearCacheIsInvalidated
+	testCases := []mapTestCase{
+		{
+			name: "Remove",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				v, err := tcx.M.Remove(ctx, i)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.Equal(tcx.T, v, i)
+			},
+		},
+		{
+			name: "RemoveIfSame",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				b, err := tcx.M.RemoveIfSame(ctx, i, i)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.True(tcx.T, b)
+			},
+		},
+	}
+	invalidationRunner(t, testCases)
+}
+
+func TestAfterDeleteNearCacheIsInvalidated(t *testing.T) {
+	// port of: com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest#testAfterDeleteNearCacheIsInvalidated
+	testCases := []mapTestCase{
+		{
+			name: "Delete",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				if err := tcx.M.Delete(ctx, i); err != nil {
+					tcx.T.Fatal(err)
+				}
+			},
+		},
+	}
+	invalidationRunner(t, testCases)
+}
+
+func TestAfterPutNearCacheIsInvalidated(t *testing.T) {
+	// port of: com.hazelcast.client.map.impl.nearcache.ClientMapNearCacheTest#testAfterPutAsyncNearCacheIsInvalidated
+	testCases := []mapTestCase{
+		{
+			name: "Put",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				if err := tcx.M.Delete(ctx, i); err != nil {
```

Comment:
This should be Put()

Reply:
Fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/806/commits/0340cb385720900db303b5811060b367f99c9526